File: workspace/scala/src/test/scala/com/dslplatform/ocd/values/TestOptDateValue.scala
package com.dslplatform.ocd.values

import OptDateInValue.OptDateValue
import com.dslplatform.api.patterns.ServiceLocator
import org.scalatest._
import scala.reflect.runtime.universe._

class TestOptDateValue extends SpecExtensions {
  def `Test property field type: Option[org.joda.time.LocalDate]`() =
    checkType(
      typeOf[Option[org.joda.time.LocalDate]]
    , typeOf[OptDateValue].member("optDate": TermName).asMethod.returnType
    )

  def `Test property default value: None`() {
    assert(OptDateValue().optDate === None)
  }
}

File: workspace/java/src/test/java/com/dslplatform/ocd/values/TestOptDateValue.java
package com.dslplatform.ocd.values;

import com.dslplatform.client.Bootstrap;
import com.dslplatform.ocd.values.OptDateInValue.OptDateValue;
import com.dslplatform.patterns.ServiceLocator;
import org.junit.*;
import static org.junit.Assert.*;

public class TestOptDateValue {
    private static ServiceLocator locator;

    @BeforeClass
    public static void setUpClass() throws Exception {
//        locator = Bootstrap.init(TestOptDateValue.class.getResourceAsStream("dsl-project.ini"));
    }

    @AfterClass
    public static void tearDownClass() throws Exception {
//        locator.resolve(java.util.concurrent.ExecutorService.class).shutdown();
//        locator = null;
    }

    @Before
    public void setUp() throws Exception {
    }

    @After
    public void tearDown() throws Exception {
    }

    @Test
    public void testFieldType() throws NoSuchFieldException {
        assertEquals(
                org.joda.time.LocalDate.class,
                OptDateValue.class.getDeclaredField("optDate").getGenericType());
    }

    @Test
    public void testGetterType() throws NoSuchMethodException {
        assertEquals(
                org.joda.time.LocalDate.class,
                OptDateValue.class.getMethod("getOptDate").getGenericReturnType());
    }

    @Test
    public void testSetterType() throws NoSuchMethodException {
        assertEquals(
                OptDateValue.class,
                OptDateValue.class.getMethod("setOptDate", org.joda.time.LocalDate.class).getReturnType());
    }

    @Test
    public void testDefaultPropertyValue() {
        assertNull(new OptDateValue().getOptDate());
    }
}
